src/main/java/seedu/contax/ui/onboarding/OnboardingUtil.java

```diff
@@ -0,0 +1,104 @@
+package seedu.contax.ui.onboarding;
+
+import seedu.contax.logic.commands.AddCommand;
+import seedu.contax.logic.commands.Command;
+import seedu.contax.logic.commands.exceptions.CommandException;
+import seedu.contax.logic.parser.AddressBookParser;
+import seedu.contax.logic.parser.exceptions.ParseException;
+import seedu.contax.model.Model;
+import seedu.contax.model.onboarding.OnboardingStep;
+import seedu.contax.model.person.Person;
+import seedu.contax.model.util.SampleDataUtil;
+
+
+/**
+ * This class provides utilities functions for the onboarding window.
+ */
+public class OnboardingUtil {
+
+    private static final AddressBookParser parser = new AddressBookParser();
+    private static final String INVALID_COMMAND = "Invalid format! "
+            + "Follow the format 'add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS'"
+            + "\n\nExample: add n/Johnny p/91234567 e/Johnny@j.com a/Johnny street";
+
+    /**
+     * Populates the given Model with the sample data
+     * @param model the Model to be populated
+     */
+    public static void populateWithSample(Model model) {
+        Person[] samplePersons = SampleDataUtil.getSamplePersons();
+        for (Person p : samplePersons) {
+            model.addPerson(p);
+        }
+    }
+
+    /**
+     * Returns the name string of the last person in the given Model
+     * @param model the UniquePersonList to be searched
+     * @return name of the last person
+     */
+    public static String getLastestPersonName(Model model) {
+        return model.getAddressBook().getPersonList()
+                .get(model.getAddressBook().getPersonList().size() - 1).getName().toString();
+    }
+
+    /**
+     * Returns the last Person in the given Model
+     * @param model the Model to be searched
+     * @return Person object of the last person
+     */
+    public static Person getLatestPerson(Model model) {
+        return model.getAddressBook().getPersonList()
+                .get(model.getAddressBook().getPersonList().size() - 1);
+    }
+
+    /**
+     * Checks if the given command is valid. Updates the label with error message if it is not.
+     * @param command command string to be validated
+     * @param instructionLabel label to print error message on
+     */
+    public static boolean isCommandValid(String command, OnboardingInstruction instructionLabel) {
+        try {
+            parser.parseCommand(command);
+            return true;
+        } catch (ParseException e) {
+            instructionLabel.setText(INVALID_COMMAND);
```

Comment:
You can decouple the UI completely if you do the `setText` in `OnboardingWindow`, since you are only calling this from a single place. `isCommandValid` should serve only 1 purpose (from its name).

Reply:
Fixed